modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/page/IntegerFieldEditor.java

```diff
@@ -26,38 +28,45 @@
  * A Field editor for integer preferences.
  * </p>
  */
-public class IntegerFieldEditor extends FieldEditor {
-
+public class IntegerFieldEditor extends FieldEditor<Integer> {
 	private static final Logger LOGGER = LoggerCreator.createLogger(IntegerFieldEditor.class);
 
 	private final TextField textField;
+	private SimpleIntegerProperty value = new SimpleIntegerProperty(this, "value");
 
 	public IntegerFieldEditor(String name, String label) {
 		super(name, label);
 		this.textField = new TextField();
 		getChildren().add(textField);
 
-		configureValidation(this.textField);
+		NodeDecorator.apply(this.textField, statusProperty());
+		this.textField.textProperty().addListener((obs, old, newValue) -> {
+			try {
+				value.set(Integer.parseInt(newValue));
```

Comment:
Does JFace do it like this? Don't we want to support eg group-separators (`.` in german, `,` in us, ...)